src/mono/IosAppBuilderTasks/IosAppBuilderTask.cs

```diff
@@ -0,0 +1,143 @@
+    /// </summary>
+    [Required]
+    public string EntryPointLib { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to a custom main.m with custom UI
+    /// A default one is used if it's not set
+    /// </summary>
+    public string? NativeMainSource { get; set; }
+
+    /// <summary>
+    /// Produce optimized binaries (e.g. use -O2 in AOT)
+    /// </summary>
+    public bool Optimized { get; set; }
+
+    /// <summary>
+    /// Target arch, can be "arm64" (device) or "x64" (simulator) at the moment
+    /// </summary>
+    [Required]
+    public string Arch { get; set; } = ""!;
+
+    /// <summary>
+    /// DEVELOPER_TEAM provisioning, needed for arm64 builds.
+    /// </summary>
+    public string? DevTeamProvisioning { get; set; }
+
+    /// <summary>
+    /// Only generate an xcode project
+    /// </summary>
+    public bool GenerateOnlyXcodeProject { get; set; }
+
+    /// <summary>
+    /// Path to *.app bundle
+    /// </summary>
+    [Output]
+    public string AppBundlePath { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to xcode project
+    /// </summary>
+    [Output]
+    public string XcodeProjectPath { get; set; } = ""!;
+
+    public override bool Execute()
+    {
+        Utils.Logger = Log;
+        bool isDevice = Arch.Equals("arm64", StringComparison.InvariantCultureIgnoreCase);
+        if (isDevice && string.IsNullOrEmpty(CrossCompiler))
+        {
+            throw new ArgumentException("arm64 arch requires CrossCompiler");
+        }
+
+        // escape spaces
+        ProjectName = ProjectName.Replace(" ", "-");
+
+        var stopWatch = Stopwatch.StartNew();
+
+        // see https://github.com/dotnet/runtime/issues/34448
+        // Can be exposed as an input argument if needed
+        string[] excludes = {"System.Runtime.WindowsRuntime.dll"};
+        string[] libsToAot = Directory.GetFiles(AppDir, "*.dll")
+            .Where(f => !excludes.Contains(Path.GetFileName(f)))
+            .ToArray();
+
+        string binDir = Path.Combine(AppDir, $"bin-{ProjectName}-{Arch}");
+        Directory.CreateDirectory(binDir);
+
+        // run AOT compilation only for devices
+        if (isDevice)
+        {
+            if (string.IsNullOrEmpty(CrossCompiler))
+                throw new InvalidOperationException("cross-compiler is not set");
+
+            AotCompiler.PrecompileLibraries(CrossCompiler, binDir, libsToAot,
+                new Dictionary<string, string>
+                {
+                    {"DOTNET_SYSTEM_GLOBALIZATION_INVARIANT", "1"},
+                    {"MONO_PATH", AppDir},
+                },
+                Optimized);
+        }
+
+        // generate modules.m
+        AotCompiler.GenerateLinkAllFile(
+            Directory.GetFiles(binDir, "*.dll.o"),
+            Path.Combine(binDir, "modules.m"));
+
+        // generate xcode project
+        XcodeProjectPath = Xcode.GenerateXCode(ProjectName, EntryPointLib, AppDir, binDir, MonoInclude, NativeMainSource);
```

Comment:
Is this required?

Reply:
It's quick and it allows to easily open the temp xcode project and debug `runtime/bcl` issues, forward all the provisioning issues to it and build apps via `xcodebuild` tool.
mono/mono used to use ninja stuff.

Reply:
If I understood it correctly this could be an extra option not enabled by default